packages/mssql/src/MsSqlPlatform.ts

```diff
@@ -248,4 +249,31 @@ export class MsSqlPlatform extends AbstractSqlPlatform {
     return false;
   }
 
+  override getOrderByExpression(column: string, direction: QueryOrder): string[] {
+    const ret: string[] = [];
+
+    switch (direction.toUpperCase()) {
+      case QueryOrder.ASC:
+        ret.push(`${column} asc`);
+        break;
+      case QueryOrder.DESC:
+        ret.push(`${column} desc`);
+        break;
+      case QueryOrder.ASC_NULLS_FIRST:
+        ret.push(`case when ${column} is null then 0 else 1 end, ${column} asc`);
+        break;
+      case QueryOrder.ASC_NULLS_LAST:
+        ret.push(`case when ${column} is null then 1 else 0 end, ${column} asc`);
+        break;
+      case QueryOrder.DESC_NULLS_FIRST:
+        ret.push(`case when ${column} is null then 0 else 1 end, ${column} desc`);
+        break;
+      case QueryOrder.DESC_NULLS_LAST:
+        ret.push(`case when ${column} is null then 1 else 0 end, ${column} desc`);
+        break;
+    }
+
+    return ret;
```

Comment:
since you never use more than the one item in the array, you don't need an array, you can just return from the cases directly. and for asc/desc you can use the default clause together.

```suggestion
switch (direction.toUpperCase()) {
  case QueryOrder.ASC:
    ret.push(`${column} asc`);
    break;
  case QueryOrder.DESC:
    ret.push(`${column} desc`);
    break;
  case QueryOrder.ASC_NULLS_FIRST:
    return `case when ${column} is null then 0 else 1 end, ${column} asc`;
  case QueryOrder.ASC_NULLS_LAST:
    return `case when ${column} is null then 1 else 0 end, ${column} asc`;
  case QueryOrder.DESC_NULLS_FIRST:
    return `case when ${column} is null then 0 else 1 end, ${column} desc`;
  case QueryOrder.DESC_NULLS_LAST:
    return `case when ${column} is null then 1 else 0 end, ${column} desc`;
  default:
    return `${column} ${direction.toLowerCase()}`;
}
```

Reply:
I had to return an array to respect Abstract class, but I kept the logic